src/Ekart/Products/product-card.js

```diff
@@ -1,41 +1,98 @@
+  const [addedToCart, setAddedToCart] = useState(false);
+  const [snackBarOpen, setSnackBarOpen] = useState(false);
+  const [snackBarMessage, setSnackBarMessage] = useState('');
+  const ratingValue = product.rating;
+  const toggleWishList = () => {
+    setWishListed(!wishListed);
+    setSnackBarOpen(true);
+    setSnackBarMessage(
+        wishListed ? 'Removed from WishList' : 'Added to WishList');
+  }
+
+  const toggleCart = () => {
+    setAddedToCart(!addedToCart);
+    setSnackBarOpen(true);
+    setSnackBarMessage(addedToCart ? 'Removed from Cart' : 'Added to Cart');
+  }
 
-const ProductCard = ({ product }) => {
   return (
-    <Card id="wd-card" >
-        <Link key={product._id} to={`/Products/${product._id}`} style={{ textDecoration: 'none' }}>
-      <CardActionArea>
-        <CardMedia
-          component="img"
-          alt={product.name}
-          height="140"
-          image={staticImage}
-        />
-        <CardContent>
-          <Typography gutterBottom variant="h5" component="div">
-            {product.name}
-          </Typography>
-          <Typography variant="body2" color="text.secondary">
-            {product.price}
-          </Typography>
-        </CardContent>
-      </CardActionArea>
-      </Link>
-      <Button className="mb-2" variant="outlined" size="small" color="primary">
-        Add to Cart
-      </Button>
-    </Card>
+      <Card className={'wd-card'}>
+        <Link key={product._id} to={`/Products/${product._id}`}
+              style={{textDecoration: 'none'}}>
+          <CardActionArea>
+            <CardMedia
+                component="img"
+                alt={product.name}
+                height="140"
+                image={staticImage}
+            />
+            <CardContent>
+              <Typography gutterBottom variant="h5" component="div">
+                {product.name}
+              </Typography>
+              <div className={'d-flex flex-row align-items-center mb-2'}>
+                <Rating
+                    size={'small'}
+                    readOnly
+                    name="read-only"
+                    value={ratingValue}
+                    precision={0.5}
+                />
+                ({ratingValue})
+              </div>
+              <Typography variant="body2" color="text.secondary">
+                {product.price}
+              </Typography>
+            </CardContent>
+          </CardActionArea>
+        </Link>
+        <div
+            className={'px-3 d-flex justify-content-between align-self-center mb-3'}>
+          {!addedToCart && <Button className="mb-0" variant="outlined"
+                                   size="small"
+                                   color="primary" onClick={toggleCart}>
+            Add to Cart
+          </Button>}
+          {addedToCart && <Button className="mb-0 wd-in-cart" variant="outlined"
+                                  size="small"
+                                  color="primary">
```

Comment:
```suggestion
                                  color="primary" onClick={toggleCart}>
```

For the cart toggle to work correctly

Reply:
Yes actually I left it like that because I wasn't sure what we are going to allow the user to do. If we give the quantity option then should we allow the user to be able to remove from the cart directly from the overview of the product list. Perhaps we can discuss more on that. I'll add this change for now.